From 5598a6c5fd9ed6e27ffb62425edcc2add390b085 Mon Sep 17 00:00:00 2001
From: Steve Cornelius <steve.cornelius@freescale.com>
Date: Wed, 18 Apr 2012 13:46:54 -0700
Subject: [PATCH 1389/2463] ENGR00180943-8: Differentiate inclusions for OF
 versus non-OF platforms

Differentiate inclusions for OF versus non-OF platforms,
and include clock control subsystems for ARM targets.

Signed-off-by: Steve Cornelius <steve.cornelius@freescale.com>
---
 drivers/crypto/caam/compat.h |   13 ++++++++++++-
 1 files changed, 12 insertions(+), 1 deletions(-)

diff --git a/drivers/crypto/caam/compat.h b/drivers/crypto/caam/compat.h
index 5585416..9ad93d4 100644
--- a/drivers/crypto/caam/compat.h
+++ b/drivers/crypto/caam/compat.h
@@ -12,7 +12,6 @@
 #include <linux/interrupt.h>
 #include <linux/crypto.h>
 #include <linux/hw_random.h>
-#include <linux/of_platform.h>
 #include <linux/dma-mapping.h>
 #include <linux/io.h>
 #include <linux/spinlock.h>
@@ -22,6 +21,18 @@
 #include <linux/types.h>
 #include <linux/debugfs.h>
 #include <linux/circ_buf.h>
+
+#ifdef CONFIG_OF
+#include <linux/of_platform.h>
+#else
+#include <linux/platform_device.h>
+#endif
+
+#ifdef CONFIG_ARM /* needs the clock control subsystem */
+#include <linux/clk.h>
+#include <asm/cacheflush.h>
+#endif
+
 #include <net/xfrm.h>
 
 #include <crypto/algapi.h>
-- 
1.7.7.4

